e2e/testcases/basic_test.go

```diff
@@ -0,0 +1,318 @@
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Replace bob with robert rolebinding")
+	nt.WaitForRepoSyncs()
+
+	if err := nt.ValidateNotFound("bob-rolebinding", "backend", &rbacv1.RoleBinding{}); err != nil {
+		nt.T.Fatal("bob-rolebinding is not deleted")
+	}
+
+	if err := nt.Validate("robert-rolebinding", "backend", &rbacv1.RoleBinding{}); err != nil {
+		nt.T.Fatal("robert-rolebinding not found")
+	}
+}
+
+func manageNamespace(nt *nomostest.NT, namespace string) {
+	nt.T.Log("Add an unmanaged resource into the namespace as a control")
+	nt.T.Log("We should never modify this resource")
+	_, err := nt.Kubectl("apply", "-f", fmt.Sprintf("%s/reserved_namespaces/unmanaged-service.%s.yaml", yamlDir, namespace))
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Add resource to manage")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/service.yaml", yamlDir), fmt.Sprintf("acme/namespaces/%s/service.yaml", namespace))
+	nt.T.Log("Start managing the namespace")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/namespace.%s.yaml", yamlDir, namespace), fmt.Sprintf("acme/namespaces/%s/namespace.yaml", namespace))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Start managing the namespace")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Wait until managed service appears on the cluster")
+	if err := nt.Validate("some-service", namespace, &corev1.Service{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Remove the namespace directory from the repo")
+	nt.RootRepos[configsync.RootSyncName].Remove(fmt.Sprintf("acme/namespaces/%s", namespace))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove the namespace from the managed set of namespaces")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Wait until the managed resource disappears from the cluster")
+	if err := nt.ValidateNotFound("some-service", namespace, &corev1.Service{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Ensure that the unmanaged service remained")
+	if err := nt.Validate("some-other-service", namespace, &corev1.Service{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	someOtherService := &corev1.Service{}
+	if err := nt.Get("some-other-service", namespace, someOtherService); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := nt.Delete(someOtherService); err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func unmanageNamespace(nt *nomostest.NT, namespace string) {
+	nt.T.Log("stop managing the system namespace")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/unmanaged-namespace.%s.yaml", yamlDir, namespace), fmt.Sprintf("acme/namespaces/%s/namespace.yaml", namespace))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Stop managing the namespace")
+	nt.WaitForRepoSyncs()
+}
+
+func TestNamespaceDefaultCanBeManaged(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation2)
+	manageNamespace(nt, "default")
+	unmanageNamespace(nt, "default")
+}
+
```

Comment:
What happened to the `gatekeeper-system` test case?

Reply:
must've missed this one. I added the gatekeeper-system test, but it works slightly different. The bats test create a yaml file and use `kubectl apply` and `kubectl delete` on the generated yaml to create the namespace. The yaml file is identical to `namespace.gatekeeper-system.yaml` so i use `mustkubectl` directly on this file instead. Let me know if you think this can be a concern